### ISSUE TITLE
Modal does not center correctly on scrolled pages

### ISSUE THREAD

**Issue by andybritton-info** [infor-design/enterprise#8887]
**Describe the bug**
When a modal is displayed on a scrolled page, it does not correctly center vertically. 

**To Reproduce**

1. Go to ['demo page'](https://stackblitz.com/edit/stackblitz-starters-o49hrw?file=styles.css,index.html)
2. Scroll down the page until the 'Show Modal' button is at least half way up the browser window
3. Click on 'Show Modal'
4. Observe that the modal is not vertically centered

**Expected behavior**
The modal should be in the middle of the browser window, regardless of the scrolling position of the page.

**Version**

- ids-enterprise: v4.96.0

**Screenshots**
![image](https://github.com/infor-design/enterprise/assets/173856238/2859d1cd-eb45-4bcf-b485-e51aa2e34b09)

**Platform**
 - Infor Application/Team Name: Infor Pathway
 - Device: Windows PC
 - OS Version: Windows 11
 - Browser Name: Chrome, Firefox

**Additional context**
Firstly, I note that the div.modal-page-container is positioned relative to the initial containing block rather than the viewport.
![image](https://github.com/infor-design/enterprise/assets/173856238/a627ccfc-164e-4170-a756-80e6e9410599)
Overriding the above with:
![image](https://github.com/infor-design/enterprise/assets/173856238/07d9852b-4def-404c-93cc-a9bec7460419)
... results in a correctly centered modal regardless of page scroll position:
![image](https://github.com/infor-design/enterprise/assets/173856238/d64f40c3-b42b-4f10-a184-7825847b6999)
The above explains the behavior but I don't know if it's good enough to propose as a fix.

Secondly, I note that this only really came to light because I had a modal with no 'focusable' controls on it and a page that was twice the height of the viewport. As long as there is a focusable control, and as long as the modal code correctly identifies it and executes .focus against it, then the browser will scroll the control (and often the entire modal) back into view. In my case, I had a modal where the only focusable controls where the buttons on the .modal-buttonset, but I had these disabled at the point the modal was being displayed. The modal code was incorrectly trying to focus one of these buttons, but as they were disabled, the browser ignored the event and did not scroll the page to bring the modal into view.

**Comment by tmcconechy** [comment]
@andybritton-info noting for future dev that it works ok on https://main-enterprise.demo.design.infor.com/components/modal/test-body-scroll.html

And perhaps that could help with a workaround until we get to it

**Comment by andybritton-info** [comment]
Ah yes - I see how that works in the demo now. thanks

**Comment by yohannahbautista** [comment]
@andybritton-info  @tmcconechy  @ericangeles 
Hello, I've been trying to reproduce the bug in IDS locally. I was able to do it by clearing everything in https://github.com/infor-design/enterprise/blob/main/app/views/layout.html and leaving only `{{> content}}` inside the `<body>` (all classes in body should be removed). The fix mentioned above works, but also given that I had to remove most of the main template for IDS, I'm not sure if a fix is needed or just extra steps to implement it in stackblitz.

I managed to make it work by adding what was in layout.html: https://stackblitz.com/edit/stackblitz-starters-ibfjke?file=index.html

**Comment by techabilla** [comment]
@yohannahbautista, perhaps it is enough to note in documentation that the modal positions itself absolutely and relies on having a non-scrollable, positioned ancestor to anchor it. That said, it might make it more difficult when introducing the component into an existing project, as was the case for me.

**Comment by tmcconechy** [comment]
@yohannahbautista i agree maybe add a note in the docs and lets go with this. We have a lot of other items to address.

**Comment by tmcconechy** [comment]
Does that means we close with the stackblitz or let QA look at it?
